common-content/en/module/tools/awk/index.md

```diff
@@ -0,0 +1,24 @@
++++
+title = "awk"
+headless = true
+time = 20
+facilitation = false
+emoji= "💻"
+[objectives]
+    1="Filter a space-separated file and extract a positional field using awk"
+    2="Sum the numbers in a column of a space-separated file using awk"
+    3="Explain what NF and $NF mean in awk"
++++
+
+`awk` is a tiny programming language for manipulating columns of data.
+
+![Julia Evans' comic about awk](https://wizardzines.com/images/uploads/awk.png)
+
+(Source, including text-only transcript: https://wizardzines.com/comics/awk/)
+
+Learn about it from its man page (and the backlog exercises).
+
+Some awk programs you should understand and be able to write:
```

Comment:
The backlog exercises cover all of these, so I was mostly leaning on those for the actual learning experience with the tools... Do you have another format in mind?

Reply:
Yeah it's about motivating attention.  You need them to stop at this point and go learn that thing and then check they've learned it. You're basically trying to be their best focus friend helping them to actually do it and not just scroll past it. So anything you can do to create that short summative assessment moment (even better if you can do formative and summative) is super helpful. It can be a really small check.